src/backend/Csrs.Api/Services/DocumentService.cs

```diff
@@ -233,26 +188,19 @@ private async Task<List<FileSystemItem>> GetListFilesInFolder(string entityId, s
         /// Initial Access Check
         /// </summary>
         /// <param name="entityId">File Id</param>
-        /// <param name="entityName">Unused currently</param>
+        /// <param name="BCeIDUserId">BCeID</param>
+        /// <param name="cancellationToken">cancellation token</param>
         /// <returns></returns>
-        private async Task<bool> CanAccessDocument(string entityId, string entityName, string BCeIDUserId, CancellationToken cancellationToken)
+        private async Task<MicrosoftDynamicsCRMssgCsrsfile> CanAccessDocument(string entityId, string BCeIDUserId, CancellationToken cancellationToken)
         {
 
-            if (String.IsNullOrEmpty(BCeIDUserId)) return false;
-
-            MicrosoftDynamicsCRMssgCsrspartyCollection partiesCollection = await _dynamicsClient.GetPartyByBCeIdAsync(BCeIDUserId, cancellationToken);
+            if (string.IsNullOrEmpty(BCeIDUserId)) return null;
 
-            if (partiesCollection is null || partiesCollection.Value.Count == 0) return false;
+            string partyId = await _dynamicsClient.GetPartyIdByBCeIdAsync(BCeIDUserId, cancellationToken);
```

Comment:
add try/catch can throw HttpOperationExecption  these calls call return a null if nothing found they throw GetSharepointDocumentLocationIdByRelatveUrl

Reply:
That is handled in the dynamic extension.